Update ldap.adoc (#39133)
keycloak/keycloak: docs/documentation/server_admin/topics/user-federation/ldap.adoc

When {project_name} updates a password, {project_name} sends the password in plain-text format. This action is different from updating the password in the built-in {project_name} database, where {project_name} hashes and salts the password before sending it to the database. For LDAP, {project_name} relies on the LDAP server to hash and salt the password.

> By default, LDAP servers such as MSAD, RHDS, or FreeIPA hash and salt passwords. Other LDAP servers such as OpenLDAP stores the passwords in plain-text unless you use the _LDAPv3 Password Modify Extended Operation_ as described in https://datatracker.ietf.org/doc/html/rfc5280#section-4.2.1.3[RFC3062]. ApacheDS can be configured to hash and salt passwords automatically when created, if the `passwordHashing Interceptor` is enabled. Enable the LDAPv3 Password Modify Extended Operation in the LDAP configuration page. See the documentation of your LDAP server for more details.

Reviewer: Slightly reordered and adjusted, as "store" refers to "Other LDAP servers". Moving ApacheDS to the end of the paragraph to to keep the block "Other LDAP server ... See documentation of your LDAP server ..." together. 

```suggestion
By default, LDAP servers such as MSAD, RHDS, or FreeIPA hash and salt passwords. Other LDAP servers such as OpenLDAP store the passwords in plain-text unless you use the _LDAPv3 Password Modify Extended Operation_ as described in https://datatracker.ietf.org/doc/html/rfc5280#section-4.2.1.3[RFC3062]. Enable the LDAPv3 Password Modify Extended Operation in the LDAP configuration page. See the documentation of your LDAP server for more details. Configure ApacheDS to hash and salt passwords automatically by enabling the passwordHashing interceptor.
```
